### PR TITLE
Google Docs Usability Audit / Improvements

### LINKED ISSUE #13912: [USABILITY] Google Docs Usability Audit

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #13912 

## Release Notes

- **New Features**
	- Enhanced Google Drive integration to monitor specific folders for new Google Docs documents.
	- Improved setup experience with sample records emitted during the initial run.
	- Users can now specify folders to watch for new documents, defaulting to the root folder.

- **Bug Fixes**
	- Updated filtering to ensure only Google Docs are processed from monitored folders.

- **Documentation**
	- Clarified functionality related to folder monitoring and document retrieval for better user understanding.

- **Chores**
	- Updated version numbers and dependencies to improve overall functionality and performance.

### CODE REVIEW

File: components/google_docs/sources/new-document-created/new-document-created.mjs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import common from "../common/base.mjs";
+
+export default {
+  ...common,
+  key: "google_docs-new-document-created",
+  name: "New Document Created (Instant)",
```
Comment (GTFalcao) [inline components/google_docs/sources/new-document-created/new-document-created.mjs:6]: Should these sources be labeled as instant? Aren't they timer-based?

Also, can we include docs links in the descriptions?

**Review by GTFalcao — COMMENTED** [review]
Looks good to go, just commented on a text/description matter, moving forward to QA

File: components/google_docs/sources/new-document-created/new-document-created.mjs (hunk 1 above)
Comment (michelle0927) [reply components/google_docs/sources/new-document-created/new-document-created.mjs:6]: - These sources are subscription based. They inherit from Google Drive, like I mentioned in the sheet. [new-files-instant](https://github.com/PipedreamHQ/pipedream/blob/master/components/google_drive/sources/new-files-instant/new-files-instant.mjs). There is a timer though to renew the subscription.
- Added doc links!

**Review by GTFalcao — APPROVED** [review]

**Pull request merged**